src/python/pants/backend/python/util_rules/pex_from_targets.py

```diff
@@ -373,14 +373,24 @@ async def create_pex_from_targets(
             chosen_resolve = await Get(
                 ChosenPythonResolve, ChosenPythonResolveRequest(request.addresses)
             )
-            loaded_lockfile = await Get(
-                LoadedLockfile, LoadedLockfileRequest(chosen_resolve.lockfile)
+            requirements, loaded_lockfile = await MultiGet(
+                Get(
+                    PexRequirements,
+                    _PexRequirementsRequest(request.addresses, resolve_name=chosen_resolve.name),
+                ),
+                Get(LoadedLockfile, LoadedLockfileRequest(chosen_resolve.lockfile)),
             )
             pex_native_subsetting_supported = loaded_lockfile.is_pex_native
             if loaded_lockfile.constraints_strings:
                 requirements = dataclasses.replace(
-                    requirements, constraints_strings=loaded_lockfile.constraints_strings
+                    requirements,
+                    constraints_strings=loaded_lockfile.constraints_strings,
                 )
```

Comment:
This means that this PR is not safe to land :( using a `pex_binary` target with `platforms` specified results in the error

> This state is not legal because it would mean there's ambiguity with which constraints to use.

I need to understand `pex_from_targets.py` better before landing this.